marlowe/security.md

```diff
@@ -0,0 +1,112 @@
+
+The input being applied to the contract in the transaction is defined in the Plutus `Redeemer` associated with spending the UTxO from the Marlowe script address, along with the slot validity interval for the transaction specified in the transaction body. The input is a sequence of zero or more deposits, choices, and notifications. The consequences of applying this input to the contract can be studied using a tool like Marlowe Playground or `marlowe-cli run prepare`.
+
+The `Redeemer` can be extracted from the unsigned transaction body and deserialized to [`Language.Marlowe.Scripts.MarloweInput`](src/Language/Marlowe/Scripts.hs) using the function `Plutus.V2.Ledger.Api.fromData`. The command-line tool `marlowe-cli util slotting` will compute the relationship between the slots mentioned in the validity interval to the POSIX times in the contract.
+
+
+#### What else is occurring in the transaction?
+
+The unsigned transaction may contain other spending and payments beyond that specified for the Marlowe contract. This can be examined with the tool `cardano-cli transaction view`.
+
+
+### Marlowe Payout Validator
+
+In addition to the *semantics validator* discussed above, Marlowe has a *payout validator*. The same security consideration hold for it. Its script hash can be computed as follows:
+```bash
+nix run 'github:input-output-hk/marlowe-cardano#marlowe-cli' -- role validator --mainnet --out-file /dev/null --print-hash
+addr1w9yswm4tyqjrmj2xy5glhx9fe7m3n7rwj6fz3qfekly3mucd3rynq
+```
+```console
+addr1w9yswm4tyqjrmj2xy5glhx9fe7m3n7rwj6fz3qfekly3mucd3rynq
+Validator hash: "49076eab20243dc9462511fb98a9cfb719f86e9692288139b7c91df3"
+```
+
+Deserialization of the `Datum` and `Redeemer` should target `(Plutus.V2.Ledger.Api.CurrencySymbol, Plutus.V2.Ledger.Api.TokenName)` and `()`, respectively.
+
+
+## Marlowe Runtime
+
+[Marlowe Runtime](../marlowe-runtime/) provides off-chain services that discover Marlowe contract history and build transactions that apply input to Marlowe contracts. The Marlowe Runtime deployment does not have to be trusted if one carefully examines the Marlowe transactions it produces, as discussed in the previous section. If Marlowe Runtime is deployed as part of a web service, then one needs to be aware of the possibility of person-in-the-middle, cross-site scripting, and other attacks.
+
+Marlowe Runtime contains a [registry of known Marlowe script versions](../marlowe-runtime/src/Language/Marlowe/Runtime/Core/ScriptRegistry.hs) that it uses to create new Marlowe transactions. One should only use Marlowe Runtime to create contracts if one trusts the script hashes in that registry. The Marlowe test suite verifies that [the script registry has not be inadvertently or maliciously altered](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe-runtime/test/Language/Marlowe/Runtime/Core/ScriptRegistrySpec.hs). (Of course, this does not guarantee that the test itself has not been altered.
```

Comment:
Suggest changing "be" to "been" in the link: 

"The Marlowe test suite verifies that [the script registry has not been inadvertently or maliciously altered]"

Reply:
Fixed in 8a6b693d0d03bd91478b2bc4666c268edefdbd7d.